src/crystal/system/unix/process.cr

```diff
@@ -343,15 +343,7 @@ struct Crystal::System::Process
     reopen_io(output, ORIGINAL_STDOUT)
     reopen_io(error, ORIGINAL_STDERR)
 
-    ENV.clear if clear_env
-    env.try &.each do |key, val|
-      if val
-        ENV[key] = val
-      else
-        ENV.delete key
-      end
-    end
-
+    LibC.environ = Env.make_envp(env, clear_env)
```

Comment:
Shouldn't that be `envp = ENV.make_envp(env, clear_env)` now? Then pass `envp` to `execvpe` where only the fallback will replace `LibC.environ`?

Reply:
That would make the specs more complicated because it would partially fix the behaviour on platforms with `LibC.execvpe`:

https://github.com/crystal-lang/crystal/blob/7572ca25334002e14f15df4e27aa959a2f3dad08/spec/std/process_spec.cr#L396-L403 

So my plan was to first merge a proper `execvpe` fallback implementation, then switch to passing the environment pointer instead of assigning `environ`.